src/sinks/datadog/traces/apm_stats/flusher.rs

```diff
@@ -0,0 +1,173 @@
+                // no sense proceeding if no payloads to flush
+                None
+            } else {
+                let payload = StatsPayload {
+                    agent_hostname: aggregator.get_agent_hostname(),
+                    agent_env: aggregator.get_agent_env(),
+                    stats: client_stats_payloads,
+                    agent_version: aggregator.get_agent_version(),
+                    client_computed: false,
+                };
+
+                Some((payload, aggregator.get_api_key()))
+            }
+        } {
+            if let Err(e) = self.compress_and_send(payload, api_key).await {
+                // TODO emit an internal `Error` event here, probably
+                error!(
+                    message = format!("Error while encoding and sending APM stats payloads: {}", e),
+                    internal_log_rate_limit = true,
+                );
+            }
+        }
+    }
+
+    async fn compress_and_send(
+        &self,
+        payload: StatsPayload,
+        api_key: Arc<str>,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        let (metadata, compressed_payload) = self.build_apm_stats_request_data(api_key, payload)?;
+
+        let request_metadata = RequestMetadata::new(0, 0, 0, 0, 0);
+        let trace_api_request = build_request(
+            (metadata, request_metadata),
+            compressed_payload,
+            self.compression,
+            &self.endpoint_configuration,
+        );
+
+        let http_request = trace_api_request
+            .into_http_request()
+            .context(BuildRequestSnafu)?;
+
+        self.client.send(http_request).await?;
+
+        Ok(())
+    }
+
+    fn build_apm_stats_request_data(
+        &self,
+        api_key: Arc<str>,
+        payload: StatsPayload,
+    ) -> Result<(DDTracesMetadata, Bytes), RequestBuilderError> {
+        let encoded_payload =
+            rmp_serde::to_vec_named(&payload).map_err(|e| RequestBuilderError::FailedToEncode {
+                message: "APM stats encoding failed.",
+                reason: e.to_string(),
+                dropped_events: 0,
+            })?;
+        let uncompressed_size = encoded_payload.len();
+        let metadata = DDTracesMetadata {
+            api_key,
+            endpoint: DatadogTracesEndpoint::APMStats,
+            finalizers: EventFinalizers::default(),
+            uncompressed_size,
+            content_type: "application/msgpack".to_string(),
+        };
+
+        let mut compressor = Compressor::from(self.compression);
+        match compressor.write_all(&encoded_payload) {
+            Ok(()) => {
+                let bytes = compressor.into_inner().freeze();
+
+                Ok((metadata, bytes))
+            }
+            Err(e) => Err(RequestBuilderError::FailedToEncode {
+                message: "APM stats payload compression failed.",
+                reason: e.to_string(),
+                dropped_events: 0,
```

Comment:
Probably a philosophical question here: why shouldn't we consider the APM stats just as valuable as straight-up metrics events, and thus worth of tracking as dropped?

Reply:
It's a bit ironic actually. The original code before this change, was sending the stats payloads with the trace payloads and thus the trace payloads were being included in BytesSent etc. But the dropped events was set to 0 here, as you note ...

After discussing with @jszwedko we were of the mind that these stats payloads are not actual trace events and so should be not included in BytesSent etc.

Applying that logic to this PR's changes and I think that means we keep it as not counting it as dropped events.

But, curious your thoughts on it with that added context now.

Reply:
I'm fine with that explanation.